### PR TITLE
[RHOAIENG-289] Emit metadata from e2e and gitops pipeline

### PR DESCRIPTION
**Pull request by piotrpdev**
## Description

Changes the GitOps pipeline and the new E2E pipeline to emit metadata as [Pipeline Results](https://tekton.dev/docs/pipelines/pipelines/#emitting-results-from-a-pipeline).

See [\[RHOAIENG-289\]](https://issues.redhat.com/browse/RHOAIENG-289) for more info.

## How Has This Been Tested?

OpenShift Local/CRC on my machine, with both git and s3, on my fork. I haven't tested if the old pipelines still work but they should since they use separate tasks.

> [!NOTE]
> If you plan on testing this PR, you will need to add this parameter to the GitOps PipelineRun:
> ```yaml
>   - name: useMergedBuildTestPipeline
>     value: "true"
> ```
> This parameter can be deleted from the pipeline altogether once https://github.com/opendatahub-io/ai-edge/pull/177#issuecomment-1851956873 is resolved.

<details><summary>Typical E2E pipeline results</summary>

```yaml
  pipelineResults:
    - name: git-model-fetched-commit
      value: f89e2bc61e6ff7f539c52fded3a2bdc991a55b7c
    - name: git-model-fetched-url
      value: 'https://github.com/piotrpdev/ai-edge.git'
    - name: git-model-fetched-commit-epoch
      value: '1702404808'
    - name: git-containerfile-fetched-commit
      value: f89e2bc61e6ff7f539c52fded3a2bdc991a55b7c
    - name: git-containerfile-fetched-url
      value: 'https://github.com/piotrpdev/ai-edge'
    - name: git-containerfile-fetched-commit-epoch
      value: '1702404808'
    - name: model-files-size
      value: |
        2084
    - name: model-files-list
      value: |
        MLmodel
        README.md
        conda.yaml
        convert_csv_to_json.py
        dataset.csv
        python_env.yaml
        requirements.txt
        tf2model/
    - name: internal-registry-url
      value: >
        image-registry.openshift-image-registry.svc:5000/pplaczek-pipeline-dev/tensorflow-housing:1
    - name: target-registry-url
      value: quay.io/pplaczek/tensorflow-housing
    - name: internal-image-url
      value: >-
        image-registry.openshift-image-registry.svc:5000/pplaczek-pipeline-dev/tensorflow-housing@sha256:0cc9a636c2f18b0f15224a234995ecd27a3dc2e5eb7ffefc8ecfd72c099da31f
    - name: target-image-url
      value: >-
        quay.io/pplaczek/tensorflow-housing:1-fa9e93e0-c66c-4075-9333-61769420f102
    - name: internal-image-size
      value: '126507187'
    - name: buildah-sha
      value: 'sha256:0cc9a636c2f18b0f15224a234995ecd27a3dc2e5eb7ffefc8ecfd72c099da31f'
    - name: model-name
      value: tensorflow-housing
    - name: model-version
      value: '1'
    - name: internal-image-created-at
      value: '2023-12-12T18:39:25Z'
    - name: internal-image-buildah-version
      value: 1.24.2
```

</details>

<details><summary>Typical GitOps pipeline results</summary>

```yaml
  pipelineResults:
    - name: target-registry-url
      value: quay.io/pplaczek/tensorflow-housing
    - name: image-sha
      value: 'sha256:0cc9a636c2f18b0f15224a234995ecd27a3dc2e5eb7ffefc8ecfd72c099da31f'
    - name: pr-url
      value: 'https://github.com/piotrpdev/ai-edge/pull/5'
```

</details>

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pipelines/tekton/aiedge-e2e/aiedge-e2e.pipelinerun.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ spec:
   - name: git-revision
     value: "main"
   - name: test-endpoint
-    value: "invocations"
```
Comment (LaVLaS) [inline pipelines/tekton/aiedge-e2e/aiedge-e2e.pipelinerun.yaml:32]: Was this change intentional since the `invocations` endpoint was used for the `bike-rentals-auto-ml` model?

**Comment by LaVLaS** [comment]
Deploying locally worked for me with all of the pipeline results accurately listed upon successful execution.  Just have one question about the `test-endpoint` change in the PipelineRun

File: pipelines/tekton/aiedge-e2e/aiedge-e2e.pipelinerun.yaml (hunk 1 above)
Comment (piotrpdev) [reply pipelines/tekton/aiedge-e2e/aiedge-e2e.pipelinerun.yaml:32]: Whoops, must've made a mistake while rebasing :sweat_smile:. Going to revert this.

**Review by LaVLaS — CHANGES_REQUESTED** [review]
Last request, can you update the README(s) to reflect that the PipelineRun will generate results from the respective PipelineRuns

**Comment by piotrpdev** [comment]
@LaVLaS done

**Review by LaVLaS — APPROVED** [review]
